src/layout/header/HeaderUserProfile.tsx

```diff
@@ -0,0 +1,51 @@
+import { FC, useEffect } from 'react';
+import { ActionIcon, Avatar, Flex, Group, Text } from '@mantine/core';
+import { PROFILE_ROUTE } from 'utils/constsRoutes';
+import { useNavigate } from 'react-router-dom';
+import { useAppDispatch, useAppSelector } from 'hooks/redux-hooks';
+import { getUser, loadUserAvatar } from 'redux/slice/userSlice';
+import { IconCoins } from '@tabler/icons-react';
+
+export const HeaderUserProfile: FC = () => {
+  const dispatch = useAppDispatch();
+  const user = useAppSelector(getUser);
+
+  useEffect(() => {
+    if (user.id) {
+      dispatch(loadUserAvatar(user.id));
+    }
+  }, [dispatch, user.id]);
+
+  const navigate = useNavigate();
+  const handleProfileClick = () => {
+    navigate(PROFILE_ROUTE);
+  };
+
```

Comment:
place the hooks on top, useEffect above the return.
add empty lines for hooks with different meanings, for example: react, redux, router, custom